pkg/scheduler/scheduler.go

```diff
@@ -761,43 +737,62 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	}()
 }
 
-type podConditionUpdaterImpl struct {
-	Client clientset.Interface
-}
-
-func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
+// update updates the condition of a pod based on the passed PodCondition
+// used only in case of scheduling errors. If we succeed with scheduling,
+// PodScheduled condition will be updated in apiserver in /bind
+// handler so that binding and setting PodCondition it is atomic.
+func (sched *Scheduler) update(pod *v1.Pod, condition *v1.PodCondition) error {
+	// This should only happen in unit tests
```

Comment:
I do not know if it's a good option to have code just for the tests within the method implementation. Can it be handled from the tests itself?

Reply:
@zouyee you can mock the kube client with `k8s.io/client-go/kubernetes/fake.NewSimpleClientset()` and inject predefined objects